api/v1beta2/network_types.go

```diff
@@ -1002,3 +1027,58 @@ func (z ZoneType) String() string {
 func (z ZoneType) Equal(other ZoneType) bool {
 	return z == other
 }
+
+// ElasticIPPool allows configuring a Elastic IP pool for resources allocating
+// public IPv4 addresses on public subnets.
+type ElasticIPPool struct {
+	// PublicIpv4Pool sets a custom Public IPv4 Pool used to create Elastic IP address for resources
+	// created in public IPv4 subnets. Every IPv4 address, Elastic IP, will be allocated from the custom
+	// Public IPv4 pool that you brought to AWS, instead of Amazon-provided pool. The public IPv4 pool
+	// resource ID starts with 'ipv4pool-ec2'.
+	//
+	// +kubebuilder:validation:MaxLength=30
+	// +kubebuilder:validation:XValidation:rule="self.startsWith('ipv4pool-ec2-')",message="publicIpv4Pool must start with 'ipv4pool-ec2-'"
```

Comment:
This validation is present as an XValidation rule and in the webhook; do we need both?